templates/deployment.yaml

```diff
@@ -114,18 +210,28 @@ spec:
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
       volumes:
-      {{- if .Values.storage.nfs.enabled }}
         - name: vault-token
+      {{- if empty .Values.storage.existingPvc }}
```

Comment:
I believe an easier logic would be:
- if exitingPvc is defined, then use it 
- elseif nfs is enabled, then use it
- else, do nothing

Reply:
Ok, Well you mean the inverted case ?
- if existingPvc claimName = existingPvc
- elif nfs claimName = nfsPvc
- else emptyDir

Reply:
Fine like that